deploy/helm/kubecf/assets/scripts/jobs/cf-cli-6-linux/setup_internal_security_group.sh

```diff
@@ -3,60 +3,218 @@
 # Never use -x or -o xtrace here because of security reasons.
 set -o errexit -o nounset -o pipefail
 
-export PATH="${CF_CLI_PATH}:${PATH}"
-
-# Write the CF_API_CA_CERT environment variable to a PEM file and set SSL_CERT_FILE pointing to it.
+# Write the CF_API_CA_CERT and UAA_CA_CERT environment variable contents to PEM files.
 if [ "${CF_API_CA_CERT:-}" != "" ]; then
     cf_api_ca_cert_pem="${DATA_DIR}/cf_api_ca_cert.pem"
     echo -n "${CF_API_CA_CERT}" > "${cf_api_ca_cert_pem}"
-    export SSL_CERT_FILE="${cf_api_ca_cert_pem}"
+    uaa_ca_cert_pem="${DATA_DIR}/uaa_ca_cert.pem"
+    echo -n "${UAA_CA_CERT}" > "${uaa_ca_cert_pem}"
 fi
 
+# wait_for_endpoint waits for an endpoint to be ready. It fails if the error is related to the
+# certificate.
+function wait_for_endpoint() {
+    local ca_cert
+    ca_cert="$1"
+    local endpoint
+    endpoint="$2"
+    while true; do
+        set +o errexit
```

Comment:
We should probably comment on why we have this weird error handling.

Reply:
Fixed with https://github.com/cloudfoundry-incubator/kubecf/pull/485/commits/c1611fe473f0cf46e1295c8fd7fad20cdf87a579.